src/libs/lodepng/lodepng.c

```diff
@@ -5349,14 +5348,23 @@ unsigned lodepng_decode(unsigned char ** out, unsigned * w, unsigned * h,
             return 56; /*unsupported color mode conversion*/
         }
 
-        outsize = lodepng_get_raw_size(*w, *h, &state->info_raw);
-        *out = (unsigned char *)lodepng_malloc(outsize);
-        if(!(*out)) {
+        lv_draw_buf_t * new_buf = lv_draw_buf_create(*w, *h, LV_COLOR_FORMAT_ARGB8888, 4 * *w);
+        if(new_buf == NULL) {
             state->error = 83; /*alloc fail*/
+            return state->error;
         }
-        else state->error = lodepng_convert(*out, data, &state->info_raw,
+
+        state->error = lodepng_convert(new_buf->data, old_buf->data, &state->info_raw,
                                                 &state->info_png.color, *w, *h);
-        lodepng_free(data);
+
+        if (state->error) {
+            lv_draw_buf_destroy(new_buf);
```

Comment:
In this case, new_buf is freed, and old_buf is freed later, but out value is not NULL.
should we set it to NULL like *out = NULL ?

Reply:
Yes, you are correct!